### PR TITLE
Add array ports to the generator

### PR DESCRIPTION
**Pull request by ibriano**
And related fixes needed to make that work.

### CODE REVIEW

File: src/bin/sol-fbp-generator/type-store.c [hunk 1]
```diff
@@ -157,6 +159,30 @@ get_string(struct sol_json_token *token)
     return strndup(token->start + 1, token->end - token->start - 2);
 }
 
+static int
+get_int(struct sol_json_token *token, int *value)
+{
+    int v;
+    char *str, *endptr = NULL;
+
+    str = strndupa(token->start, token->end - token->start);
```
Comment (lpereira) [inline src/bin/sol-fbp-generator/type-store.c]: There's no need to copy the string: call `strtol()`, and check if `endptr` points to `token->end` (if not, there was a parsing error).

**Comment by ibriano** [comment]
Comments fixed locally, if more things arise I'll send a v2

File: src/bin/sol-fbp-generator/type-store.c (hunk 1 above)
Comment (barbieri) [inline src/bin/sol-fbp-generator/type-store.c]: This is not right if JSON is malformed it may cause out of bounds access. strtol() relies on trailing \0

**Comment by ricardotk** [comment]
+1

**Comment by ibriano** [comment]
Merged.

**Pull request closed without merging**